vertx-sql-client/src/main/java/io/vertx/sqlclient/SqlCredentialsProvider.java

```diff
@@ -0,0 +1,36 @@
+package io.vertx.sqlclient;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+
+public interface SqlCredentialsProvider {
+
+  class Static implements SqlCredentialsProvider {
+
+    private final Credentials credentials;
+
+    public Static(String username, String password) {
+      this.credentials = new Credentials(username, password);
+    }
+
+    @Override
+    public Future<Credentials> getCredentials(Context context) {
+      return Future.succeededFuture(credentials);
+    }
+  }
+
+  class Credentials {
```

Comment:
For a working integration with codegen, this should be annotated with `@DataObject` and moved to its own file.

Reply:
Ok.